src/classes/PREQ_PreventPPlanParent_TEST.cls

```diff
@@ -0,0 +1,144 @@
+/*
+    Copyright (c) 2019, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Plan Requirement
+* @group-content ../../ApexDocContent/PlanRequirements.htm
+* @description Tests for PREQ_PreventPPlanParent_TDTM.
+*/
+@isTest
+private class PREQ_PreventPPlanParent_TEST {
+
+    @isTest
+    public static void pReqInsertChildPRWithSamePP() {
```

Comment:
@xinnanlisf I also think we might want to consider adding a few additional unit tests to test additional configurations:
- A unit test that has valid data (to confirm that no exception is thrown)
- A unit test that has invalid data, but the custom setting is disabled (to confirm that no exception is thrown)

Reply:
That's probably a good idea. We should get in the habit of doing this for all of our functionality. Testing good and bad data with settings toggled on/off.

Reply:
I am hesitating about the first one. Seems a bit overtesting to me. We might end up having duplicate test methods eventually.
The second one sounds good to me!

Reply:
When it comes to testing, some would say that you can't "over test" -- that said, it does complete the happy path of things working as expected. I'd say if you have time, to add it. I am curious though what you mean when you talk about having duplicate test methods though. Care to elaborate a bit?

Reply:
@lifewithryan yeah I will add it to unit tests then.

I feel like I confuse myself a bit here :). My initial thought was: The first test method Bobby purposed is in fact testing if a nested plan requirement can have look up to a program plan. (Or, if a plan requirement can have both plan requirement and program plan lookup populated) Isn't it testing against core functionality?

Reply:
Yes it's testing against core functionality simply because we're just trying to save the record and not throw an exception. It seems simple enough and this would likely never fail, but because we are adding some logic around it, its more of a "cover our butts" kind of thing. If we ever see that fail we know we really screwed something up. I feel like you can't over test, some would argue but I say "nay nay" -- I'm fine with such tests.